src/benchmarks/micro/corefx/System.Text.Json/Serializer/WriteJson.Dictionary.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public class WriteDictionary
+    {
+        private Dictionary<string, string> _dict = new Dictionary<string, string>() { { "Hello", "World" }, { "Hello2", "World2" } };
+        private static IDictionary<string, string> _iDict = new Dictionary<string, string>() { { "Hello", "World" }, { "Hello2", "World2" } };
+        private IReadOnlyDictionary<string, string> _iReadOnlyDict = new Dictionary<string, string>() { { "Hello", "World" }, { "Hello2", "World2" } };
+
+        private static ImmutableDictionary<string, string> _immutableDict;
+        private static IImmutableDictionary<string, string> _iimmutableDict;
+        private static ImmutableSortedDictionary<string, string> _immutableSortedDict;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            _immutableDict = ImmutableDictionary.CreateRange(_dict);
+            _iimmutableDict = ImmutableDictionary.CreateRange(_dict);
+            _immutableSortedDict = ImmutableSortedDictionary.CreateRange(_dict);
+        }
+
+        [BenchmarkCategory(Categories.CoreFX, Categories.JSON)]
+        [Benchmark]
+        public byte[] SerializeDict_ToUtf8Bytes()
+        {
+            return JsonSerializer.ToUtf8Bytes(_dict);
+        }
+
+        [BenchmarkCategory(Categories.CoreFX, Categories.JSON)]
+        [Benchmark]
+        public byte[] SerializeIDict_ToUtf8Bytes()
+        {
+            return JsonSerializer.ToUtf8Bytes(_iDict);
+        }
+
+        [BenchmarkCategory(Categories.CoreFX, Categories.JSON)]
+        [Benchmark]
+        public byte[] SerializeIReadOnlyDict_ToUtf8Bytes()
+        {
+            return JsonSerializer.ToUtf8Bytes(_iReadOnlyDict);
+        }
+
+        [BenchmarkCategory(Categories.CoreFX, Categories.JSON)]
+        [Benchmark]
+        public string SerializeDict_ToString()
+        {
+            return JsonSerializer.ToString(_dict);
+        }
+
+        [BenchmarkCategory(Categories.CoreFX, Categories.JSON)]
+        [Benchmark]
+        public string SerializeIDict_ToString()
+        {
+            return JsonSerializer.ToString(_iDict);
```

Comment:
In the interest of avoiding test bloat (and only testing a single dimension), it's probably not useful to add the `ToString()` tests for all the specialized test cases (like WriteJson.Dictionary), when we have the `ToUtf8Bytes` tests already. I think the overhead of transcoding the end result to UTF-16 is well understood (and captured by the existing tests).